posthog/async_migrations/utils.py

```diff
@@ -74,34 +74,40 @@ def execute_op_clickhouse(
     client._request_information = None
 
 
-def execute_on_each_shard(sql: str, args=None, settings=None) -> None:
+# returns the results of all tasks in an list where l[x] = rows per shard, l[x][y] = row in shard, l[x][y][z] = column in row in shard
+def execute_on_each_shard(sql: str, args=None, settings=None) -> List[List[Tuple[Any]]]:
```

Comment:
I suggest reverting this unless it's used. The returning structure is tricky and more code to maintain. We can save the code somewhere (e.g. https://github.com/PostHog/scratchpad/) if needed and bring it back if it's ever useful.

Reply:
Fair